examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -218,8 +221,25 @@ function mergeBufferGeometries( geometries, useGroups = false ) {
 
 			}
 
-			mergedGeometry.addGroup( offset, count, i );
+			if ( geometry.groups.length > 0 ) {
+
+				// Keep existing groups
+
+				for ( let group of geometry.groups ) {
+
+					let offsetMaterialIndex = groupIndex + group.materialIndex;
+					mergedGeometry.addGroup( offset + group.start, group.count, offsetMaterialIndex );
+					nextGroupIndex = Math.max( nextGroupIndex, offsetMaterialIndex );
+
+				}
+
+			} else {
+
+				mergedGeometry.addGroup( offset, count, i );
```

Comment:
I'm still thinking through this code - but shouldn't the "material index" parameter here be driven by `nextGroupIndex` instead of `i`?

Reply:
Yes.. I've changed it to `groupIndex` (Set to `nextGroupIndex` at the start of the outer for loop)